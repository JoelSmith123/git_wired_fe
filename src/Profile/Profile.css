.profile-select-options {
  background-color: #4c4c4c;
  height: 200px;
  margin: 40px auto;
  text-align: center;
  width: 80%;
}

.profile-select-options-title {
  color: #fff;
  padding-top: 60px;
}

.profile-select-options-btn-container {
  margin-top: 50px;
}

.recent-projects-container {
  background-color: #4c4c4c;  
  height: 600px;
  margin: 0 auto 70px;
  text-align: center;  
  width: 80%;
}

.recent-projects-container-title {
  color: #fff;
  padding-top: 20px;
}