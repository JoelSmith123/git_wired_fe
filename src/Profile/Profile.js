import React, { Component } from 'react';
// import 'Profile.css';

export default class Profile extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div className='Profile'>
        <div className='profile-select-options'>
          <h2>Pick a new wireframe!</h2>
          <button>Repo</button>
          <button>Project</button>
          <button>Type of page</button>
        </div>
        <div className='recent-projects-container'>
          <h2>Recent Projects</h2>
          <div className='recent-projects-card-container'>
            <div className='recent-projects-card'></div>
            <div className='recent-projects-card'></div>
            <div className='recent-projects-card'></div>
            <div className='recent-projects-card'></div>
          </div>
        </div>
      </div>
    )
  }
}