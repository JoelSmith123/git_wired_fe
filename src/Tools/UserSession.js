export default class UserSession {
  constructor() {
    this.tokenKey = 'userToken'
  }

  setGitWiredToken(token) {
    sessionStorage.setItem(this.tokenKey, token);
  }

  getGitWiredToken() {
    return sessionStorage.getItem(this.tokenKey);
  }

  logout() {
    sessionStorage.removeItem(this.tokenKey);
  }


  isLoggedIn = () => {
    return Boolean( this.getGitWiredToken() )
  }




  // Make Fetch /login call
  // receive userToken

  // TO DO - TEST ME
  //  - CONFIRM endpoint
  //  - CONFIRM RESPONSE STRUCTURE
  // loginUser() {
  //   let url = 'https://git-wired-be.herokuapp.com/login'
  //   fetch(url)
  //     .then(response => response.json())
  //     .then(data     => this.getToken(data).bind(this))
  //     .then(token    => this.setGitWiredToken(token).bind(this))
  //     .catch(error   => { console.log(error); });
  // }
  //
  // getToken(data) {
  //   let token = data[this.tokenKey]
  //   return token
  // }


}
