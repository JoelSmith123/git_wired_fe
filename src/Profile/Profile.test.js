import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Profile from './Profile.js'
import UserSession from '../Tools/UserSession.js'

describe('Profile', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Profile user={ new UserSession } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Profile user={new UserSession}/>)
    expect(wrapper).toMatchSnapshot()
  })  

  xit('renders RecentProjects', () => {
    // TO DO - TEST ME
  })

  xit('has functional dropdown for Repos', () => {
    // TO DO - TEST ME
  })

  xit('has functional dropdown for Projects', () => {
    // TO DO - TEST ME
  })

  xit('has functional dropdown for Type of page', () => {
    // TO DO - TEST ME
  })

  xit('can start a new project and move to WireframeEditView', () => {
    // TO DO - TEST ME
  })


});
