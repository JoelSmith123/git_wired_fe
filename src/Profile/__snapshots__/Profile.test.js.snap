// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    user={
      UserSession {
        "getURLToken": [Function],
        "isLoggedIn": [Function],
        "login": [Function],
        "tokenKey": "userToken",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="profile-select-options"
        >
          <h2
            className="profile-select-options-title"
          >
            Pick a new wireframe!
          </h2>
          <div
            className="profile-select-options-btn-container"
          >
            <button>
              Repo
            </button>
            <button>
              Project
            </button>
            <button>
              Type of page
            </button>
          </div>
        </div>,
        <div
          className="recent-projects-container"
        >
          <h2
            className="recent-projects-container-title"
          >
            Recent Projects
          </h2>
          <div
            className="recent-projects-card-container"
          >
            <RecentProjects />
          </div>
        </div>,
      ],
      "className": "Profile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="profile-select-options-title"
            >
              Pick a new wireframe!
            </h2>,
            <div
              className="profile-select-options-btn-container"
            >
              <button>
                Repo
              </button>
              <button>
                Project
              </button>
              <button>
                Type of page
              </button>
            </div>,
          ],
          "className": "profile-select-options",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Pick a new wireframe!",
              "className": "profile-select-options-title",
            },
            "ref": null,
            "rendered": "Pick a new wireframe!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button>
                  Repo
                </button>,
                <button>
                  Project
                </button>,
                <button>
                  Type of page
                </button>,
              ],
              "className": "profile-select-options-btn-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Repo",
                },
                "ref": null,
                "rendered": "Repo",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Project",
                },
                "ref": null,
                "rendered": "Project",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Type of page",
                },
                "ref": null,
                "rendered": "Type of page",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="recent-projects-container-title"
            >
              Recent Projects
            </h2>,
            <div
              className="recent-projects-card-container"
            >
              <RecentProjects />
            </div>,
          ],
          "className": "recent-projects-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Recent Projects",
              "className": "recent-projects-container-title",
            },
            "ref": null,
            "rendered": "Recent Projects",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RecentProjects />,
              "className": "recent-projects-card-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="profile-select-options"
          >
            <h2
              className="profile-select-options-title"
            >
              Pick a new wireframe!
            </h2>
            <div
              className="profile-select-options-btn-container"
            >
              <button>
                Repo
              </button>
              <button>
                Project
              </button>
              <button>
                Type of page
              </button>
            </div>
          </div>,
          <div
            className="recent-projects-container"
          >
            <h2
              className="recent-projects-container-title"
            >
              Recent Projects
            </h2>
            <div
              className="recent-projects-card-container"
            >
              <RecentProjects />
            </div>
          </div>,
        ],
        "className": "Profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="profile-select-options-title"
              >
                Pick a new wireframe!
              </h2>,
              <div
                className="profile-select-options-btn-container"
              >
                <button>
                  Repo
                </button>
                <button>
                  Project
                </button>
                <button>
                  Type of page
                </button>
              </div>,
            ],
            "className": "profile-select-options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pick a new wireframe!",
                "className": "profile-select-options-title",
              },
              "ref": null,
              "rendered": "Pick a new wireframe!",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button>
                    Repo
                  </button>,
                  <button>
                    Project
                  </button>,
                  <button>
                    Type of page
                  </button>,
                ],
                "className": "profile-select-options-btn-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Repo",
                  },
                  "ref": null,
                  "rendered": "Repo",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Project",
                  },
                  "ref": null,
                  "rendered": "Project",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Type of page",
                  },
                  "ref": null,
                  "rendered": "Type of page",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="recent-projects-container-title"
              >
                Recent Projects
              </h2>,
              <div
                className="recent-projects-card-container"
              >
                <RecentProjects />
              </div>,
            ],
            "className": "recent-projects-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recent Projects",
                "className": "recent-projects-container-title",
              },
              "ref": null,
              "rendered": "Recent Projects",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <RecentProjects />,
                "className": "recent-projects-card-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
